atef/config.py

```diff
@@ -157,14 +184,820 @@ def from_yaml(cls, filename: AnyPath) -> ConfigurationFile:
             serialized_config = yaml.safe_load(fp)
         return apischema.deserialize(cls, serialized_config)
 
-    def to_json(self):
-        """Dump this configuration file to a JSON-compatible dictionary."""
-        return apischema.serialize(ConfigurationFile, self, exclude_defaults=True)
+    def to_json(self):
+        """Dump this configuration file to a JSON-compatible dictionary."""
+        return apischema.serialize(ConfigurationFile, self, exclude_defaults=True)
+
+    def to_yaml(self):
+        """Dump this configuration file to yaml."""
+        init_yaml_support()
+        return yaml.dump(self.to_json())
+
+
+@dataclass
+class PreparedFile:
+    #: The data cache to use for the preparation step.
+    cache: DataCache = field(repr=False)
+    #: The corresponding configuration file information.
+    file: ConfigurationFile
+    #: The happi client instance.
+    client: happi.Client
+    #: The comparisons defined in the top-level file.
+    root: PreparedGroup
+
+    @classmethod
+    def from_config(
+        cls,
+        file: ConfigurationFile,
+        *,
+        client: Optional[happi.Client] = None,
+        cache: Optional[DataCache] = None,
+    ) -> PreparedFile:
+        """
+        Prepare a ConfigurationFile for running.
+
+        If available, provide an instantiated happi Client and a data
+        cache.  If unspecified, a configuration-derived happi Client will
+        be instantiated and a new data cache will be utilized.
+
+        The provided cache (or a new one) will be utilized for every
+        configuration/comparison in the file.
+
+        Parameters
+        ----------
+        file : ConfigurationFile
+            The configuration file instance.
+        client : happi.Client, optional
+            A happi Client instance.
+        cache : DataCache, optional
+            The data cache instance, if available.  If unspecified, a new data
+            cache will be instantiated.
+        """
+        if client is None:
+            client = happi.Client.from_config()
+
+        if cache is None:
+            cache = DataCache()
+
+        prepared_root = PreparedGroup.from_config(
+            file.root,
+            client=client,
+            cache=cache,
+            parent=None,
+        )
+        prepared_file = PreparedFile(
+            file=file,
+            cache=cache,
+            client=client,
+            root=prepared_root,
+        )
+        prepared_root.parent = prepared_root
```

Comment:
This bit is unintuitive to me- isn't this inviting some sort of infinite loop accident? What's the advantage of setting its parent to itself rather than keeping it as `None` or some other sentinel value?

Reply:
Unintuitive as it's a typo/bug - it should be `prepared_file`. The root configuration group's parent is the file.

Reply:
Thanks for catching it - resolved in [40844c6](https://github.com/pcdshub/atef/pull/126/commits/40844c6984898603523778210afdcd17ed509760)

Test suite needs some additional group tests, I think.